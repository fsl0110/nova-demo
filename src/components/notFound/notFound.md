#### NotFound

```jsx
const MemoryRouter = require("react-router").MemoryRouter;
<MemoryRouter>
  <NotFound />
</MemoryRouter>;
```
