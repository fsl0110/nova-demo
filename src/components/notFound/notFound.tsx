import React, { FC } from "react";
import { Link } from "react-router-dom";

// TODO: Router not routing to this page. Fix it!
export const NotFound: FC = () => (
  <div className="error">
    <div className="text-center">
      <h1>404</h1>
      <h2>Page not found!</h2>
      <Link to="/" className="error__button error__link">
        go to Startpage
      </Link>
    </div>
  </div>
);

export default NotFound;
