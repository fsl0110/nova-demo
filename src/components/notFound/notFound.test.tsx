import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { NotFound } from "./notFound";

it("renders toMatchSnapshot", () => {
  const mock = shallow(<NotFound />);
  expect(toJson(mock)).toMatchSnapshot();
});
