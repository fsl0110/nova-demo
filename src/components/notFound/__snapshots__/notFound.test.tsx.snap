// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders toMatchSnapshot 1`] = `
<div
  className="error"
>
  <div
    className="text-center"
  >
    <h1>
      404
    </h1>
    <h2>
      Page not found!
    </h2>
    <Link
      className="error__button error__link"
      to="/"
    >
      go to Startpage
    </Link>
  </div>
</div>
`;
